packages/http-client-java/emitter/src/emitter.ts

```diff
@@ -125,10 +126,16 @@ export async function $onEmit(context: EmitContext<EmitterOptions>) {
         options["flavor"] = "azure";
       }
     }
-    const builder = new CodeModelBuilder(program, context);
-    const codeModel = await builder.build();
 
-    if (!program.hasError() && !program.compilerOptions.noEmit) {
+    let codeModel: CodeModel | undefined;
+    try {
+      const builder = new CodeModelBuilder(program, context);
+      codeModel = await builder.build();
+    } catch (error: any) {
+      logError(program, error.message);
+    }
```

Comment:
Not relate to pageable

Just catch error (for some issue we had to stop right there) and convert to diagnostic.